device/sycl/src/utils/barrier.hpp

```diff
@@ -8,40 +8,59 @@
 #pragma once
 
 // Project include(s).
-#include "traccc/definitions/qualifiers.hpp"
+#include "traccc/device/concepts/barrier.hpp"
 
-// SYCL includes
+// SYCL include(s).
 #include <sycl/sycl.hpp>
 
-namespace traccc::sycl {
+namespace traccc::sycl::details {
 
+/// A barrier type for SYCL kernels.
+template <std::size_t DIMENSIONS = 1>
 struct barrier {
-    barrier(::sycl::nd_item<1> item) : m_item(item) {}
 
-    TRACCC_DEVICE
-    void blockBarrier() { m_item.barrier(); }
+    /// Verify that the barrier is used on a supported dimensionality.
+    static_assert(DIMENSIONS > 0 && DIMENSIONS < 4,
+                  "Only 1D, 2D, and 3D barriers are supported.");
```

Comment:
`static_assert` is _so_ C++17; let's turn this into a nice `requires` clause.

Reply:
I'll have to look up how to do that. :stuck_out_tongue: But sure, that sounds like an idea.